Gems/Atom/Asset/Shader/Code/Source/Editor/CommonFiles/Preprocessor.cpp

```diff
@@ -353,6 +357,42 @@ namespace AZ
             }
         }
 
+        bool FilesModifiedFromTemplate(const AZ::IO::Path& path)
+        {
+            auto result = AzFramework::FileFunc::FindFilesInPath(path.Native(), "*", true);
+            if (result.IsSuccess())
```

Comment:
I would go for an early return approach to simplify reading (I don't see a return true statement btw so there might be faulty logic)